File: src/logic/model/ILovingMemoryApplicationState.ts
import IAppPreferences from './IAppPreferences'
import IAudio from './IAudio'
import ICharity from './ICharity'
import IMemorializedPerson from './IMemorializedPerson'
import IPhoto from './IPhoto'
import IService from './IService'
import ISong from './ISong'
import IVideo from './IVideo'

interface ILovingMemoryApplicationState {
  preferences: IAppPreferences
  memorialized: IMemorializedPerson
  services: {
    burial: IService
    cremation: IService
    funeral: IService
    scattering: IService
  }
  charities: ICharity[]
  contentItems: {
    photos: IPhoto[]
    videos: IVideo[]
    songs: ISong[]
    audioClips: IAudio[]
  }
}

export default ILovingMemoryApplicationState

File: src-tauri/src/structs/ILovingMemoryApplicationState.rs
use crate::structs::IAppPreferences;
use crate::structs::ICharity;
use crate::structs::IMemorializedPerson;
use crate::structs::IService;
use crate::structs::UContentItem;

pub struct ILovingMemoryApplicationState {
  preferences: IAppPreferences::IAppPreferences,
  memorialized: IMemorializedPerson::IMemorializedPerson,
  burial: IService::IService,
  cremation: IService::IService,
  funeral: IService::IService,
  scattering: IService::IService,
  charities: Vec<ICharity::ICharity>,
  content_items: Vec<UContentItem::UContentItem>,
}

pub fn new() -> ILovingMemoryApplicationState {
  ILovingMemoryApplicationState {
    preferences: IAppPreferences::new(),
    memorialized: IMemorializedPerson::new(),
    burial: IService::new(),
    cremation: IService::new(),
    funeral: IService::new(),
    scattering: IService::new(),
    charities: Vec::<ICharity::ICharity>::new(),
    content_items: Vec::<UContentItem::UContentItem>::new(),
  }
}
